website/docs/cloud-docs/integrations/kubernetes/index.mdx

```diff
@@ -1,68 +1,163 @@
+  agentTokens:
+  - name: token-red
+  agentDeployment:
+   replicas: 1
+  autoscaling:
+    targetWorkspaces:
+    - name: us-west-development
+    - id: ws-NUVHA9feCXzAmPHx
+    - wildcardName: eu-development-*
+    minReplicas: 1
+    maxReplicas: 3
+```
 
-1. Create a [Kubernetes Secret](https://kubernetes.io/docs/concepts/configuration/secret/) with the Terraform Cloud API credentials.
+</CodeBlockConfig>
 
-   ```
-   kubectl -n $NAMESPACE create secret generic terraformrc --from-file=credentials
-   ```
+In the above example, the operator ensures that at least one agent pod is continuously running, and dynamically scales the number of pods up to a maximum of three based on the workload or resource demand. To do this, the operator will monitor the resource demand by observing the load of the designated workspaces, which are specified by the `name`, `id`, or through `wildcardName` patterns. When the workload decreases, the operator will scale down the number of agent pods.
 
-1. Add sensitive variables, such as your cloud provider credentials, to the workspace.
+Refer to the [agent pool API reference](/terraform/cloud-docs/integrations/kubernetes/api-reference#agentpool) for complete documentation of the `AgentPool` specification.
 
-   ```
-   kubectl -n $NAMESPACE create secret generic workspacesecrets --from-literal=secret_key=abc123
-   ```
+### Module
 
-1. Install the [Terraform Cloud Operator for Kubernetes via Helm](https://github.com/hashicorp/terraform-helm).
+The `Module` controller enforces an [API-driven Run workflow](/terraform/cloud-docs/run/api) and enables the deployment of Terraform modules within various workspaces as needed.
 
-   ```
-   helm repo add hashicorp https://helm.releases.hashicorp.com
+The following example deploys version `1.0.0` of the `hashicorp/module/random` module in the `workspace-name` workspace.
 
-   helm install --namespace ${RELEASE_NAMESPACE} hashicorp/terraform --generate-name
-   ```
+```yaml
+---
+apiVersion: app.terraform.io/v1alpha2
+kind: Module
+metadata:
+  name: my-module
+spec:
+  organization: kubernetes-operator
+  token:
+    secretKeyRef:
+      name: tfc-operator
+      key: token
+  module:
+    source: hashicorp/module/random
+    version: 1.0.0
+  workspace:
+    name: workspace-name
+  variables:
+  - name: string_length
+  outputs:
+  - name: random_string
+```
 
-1. To create a Terraform workspace, you can create a separate Helm chart to deploy the custom resource or examine these [examples](https://github.com/hashicorp/terraform-helm/tree/master/example).
+The operator passes the `string_length` variable that exists in the workspace to the module and stores the `random_string` outputs as either a Kubernetes secret or a ConfigMap. If the output is marked as "sensitive", it is stored as a Kubernetes secret, otherwise it is stored as a ConfigMap. The variables need to be accessible within the workspace as a workspace variable, workspace variable set, or project variable set.
```

Comment:
I assume the workspace is marking the variable as "sensitive"? Suggestion for active following that assumption below: 
```suggestion
The operator passes the workspace's `string_length` variable to the module and stores the `random_string` outputs as either a Kubernetes secret or a ConfigMap. If the workspace marks the output as "sensitive", the operator stores the `random_string` as a Kubernetes secret; otherwise, the operator stores it as a ConfigMap. The variables must be accessible within the workspace as a workspace variable, workspace variable set, or project variable set.
```